### ISSUE TITLE
Add List View in the Customizer Widgets screen

### ISSUE THREAD

**Issue by critterverse** [WordPress/gutenberg#33716]
Complimenting issue to https://github.com/WordPress/gutenberg/issues/32311

List view is such a huge usability improvement in the block editor — it would be great to add this feature to the Customizer Widgets screen. Ideally, we could add a "persistent" List view like the latest in the post editor but it will need some design exploration to see what can work best with the limited space available.

**Comment by critterverse** [comment]
Here’s a preliminary pass at how persistent List View might work in the Customizer. Since space is limited, I wonder if the sidebar area could be segmented vertically, with the top section featuring the List View and the bottom section containing the editor. 

![list-view](https://user-images.githubusercontent.com/77359364/132238943-6555d9d3-dceb-4309-9fb4-032e2df238b3.jpg)

Note that in order to accommodate the List View icon, the editor top bar has been reorganized to be more similar to the way the top bar appears in other editors, with the inserter icon all the way to the left. See also: https://github.com/WordPress/gutenberg/issues/34398#issuecomment-908721434

List View would be an especially great usability improvement if it could allow users to move widgets between Widget Areas more easily with drag and drop. While not currently possible, it would be nice if the user could drag a block into a collapsed section (e.g. another Widget Area):

https://user-images.githubusercontent.com/77359364/132238963-2fd3d172-d5ee-417a-925c-c90e04c25173.mov

**Comment by draganescu** [comment]
My fear is that in situations with tens of widgets the experience could be a lot different compared to the mockup (scroll, scroll, scroll).

**Comment by critterverse** [comment]
Good point @draganescu! Here's another variation that treats the List view as a drawer that slides out from the left like the Inserter. This version feels a bit more aligned with how the List view / Inserter work in the post editor (i.e. sharing the same sidebar area):

![Customizer List View](https://user-images.githubusercontent.com/77359364/133495862-1c7e99f5-9984-4471-8db1-42cd98d4ae1b.jpg)

We could update the "Add a block" text shown at the top of the inserter to match the type size of the "List view" label as shown above.

**Comment by noisysocks** [comment]
Noting for future devs that https://github.com/WordPress/gutenberg/issues/34398, https://github.com/WordPress/gutenberg/issues/33716, and https://github.com/WordPress/gutenberg/issues/34087 could be good to tackle together as they all involve reconfiguring the toolbar.
